xir/program.py

```diff
@@ -137,27 +137,44 @@ def use_floats(self) -> bool:
         return self._use_floats
 
 
+class ObservableFactor:
+    def __init__(self, name: str, params: [], wires: [int]):
```

Comment:
The syntax of these type hints look a little off. Also, we can make them slightly more complete:
```suggestion
    def __init__(self, name: str, params: Params, wires: Sequence[Wire]) -> None:
```